kinto/core/resource/__init__.py

```diff
@@ -638,8 +633,8 @@ def apply_changes(self, record, changes):
         """
```

Comment:
You need to update the `:returns:` and `:rtype:`. To be honest, I would love a great description of `applied_changes`, including what it should look like when some changes are happening to a deeply nested object.